src/GafferScene/MergeScenes.cpp

```diff
@@ -902,9 +903,8 @@ IECore::ConstPathMatcherDataPtr MergeScenes::computeSet( const IECore::InternedS
 					merged = new PathMatcherData();
 					result = merged;
 				case InputType::Other :
-					merged->writable().addPaths(
-						scene->setPlug()->getValue()->readable()
-					);
+					ConstPathMatcherDataPtr paths = scene->setPlug()->getValue();
+					merged->writable().addPaths( paths->readable() );
```

Comment:
I'd be curious to know if this part is actually necessary. My understanding is that the old code here was OK, because the lifetime of the temporary returned by `getValue()` outlives the call to `addPaths()`.

Reply:
I believe you are correct. I think I'd seen one farm job complete without error before adding this change. I wasn't certain enough to go without though, especially as the crashes were non-deterministic and the stacktrace does have `MergeScenes::computeSet` in the history.

FWIW, I like having a style convention matching the safe case, as I think it'd help avoid mistakes in future nodes (for me at least).

As an aside, I did a quick search and there is no other occurrence of `setNamesPlug()->getValue()->readable()` in Gaffer and only one other occurrence in the IE code base. There is also one other occurrence of `setPlug()->getValue()->readable()` in the IE code base (again passed as a temporary to `addPaths`).

Reply:
> FWIW, I like having a style convention matching the safe case, as I think it'd help avoid mistakes in future nodes (for me at least).

Agreed. It's easier to have a mantra of "assign to Ptr" than it is to add "unless you have thought about the C++ rules for temporaries and are sure you're correct", and the extra line of code doesn't hurt. Because the temporary does live long enough as a function argument, this isn't the first time I've been surprised that it doesn't when it's the target of a range-for.

> As an aside, I did a quick search and there is no other occurrence of setNamesPlug()->getValue()->readable() in Gaffer and only one other occurrence in the IE code base.

I did a slightly more general search for `getValue()->readable()` in Gaffer earlier and while there were a few, I satisfied myself that they were all kosher.